frontend/device/kobo/nickel_conf.lua

```diff
@@ -163,10 +176,22 @@ function NickelConf.colorSetting.set(new_color)
                                        new_color)
 end
 
+function NickelConf.autoColorEnabled.set(new_autocolor)
+    return NickelConf._write_kobo_conf(re_AutoColorEnabled,
+                                       auto_color_enabled_str,
+                                       new_autocolor)
+end
+
 dbg:guard(NickelConf.colorSetting, 'set',
           function(self, new_color)
               assert(new_color >= 1500 and new_color <= 6400,
                      "Wrong colorSetting value given!")
           end)
 
+dbg:guard(NickelConf.autoColorEnabled, 'set',
+          function(self, new_autocolor)
+              assert(type(new_autocolor) == "boolean",
+                     "Wrong autocolor value given (expected boolean)!")
```

Comment:
I find the error message slightly confusing. It's not so much a wrong value (which makes me think that, for example, it's too large) as the wrong type.